### PR TITLE
Issue 272

### LINKED ISSUE #261: Have IVT and Assignments Table populate with correct configuration data
Currently the IVT and Assignments Table is not properly reflecting the current configuration selection when you switch between AnalysisConfigurations in the left hand sidebar in analysis mode. A function is needed to update the table and modal box when a new configuration is created or clicked.

### PR DESCRIPTION
**Pull request by leshghi**
Update so that the IVT and the analysis values in the Assignments modal correctly reflect the current analysis config. Also fixed an outstanding bug in the IVT where it would not re-render the saved values after being saved/closed and instead reset the UI to a blank option. This should close #261.

### CODE REVIEW

File: leaf-ui/js/displayAnalysis.js [hunk 1]
```diff
@@ -277,10 +277,13 @@ function loadAnalysis(){
...
         console.log(analysisRequest);
     }
+    // TODO: figure out how to set it to the element of the map that will populate on top
+    currAnalysisConfig = analysisMap.get("Configuration1");
     // Set default UAL to preserve in future configs
     defaultUAL = currAnalysisConfig.userAssignmentsList;
+    analysisRequest = currAnalysisConfig.analysisRequest;
     // Refresh the sidebar to include the config vars
-    refreshAnalysisBar();
+    refreshAnalysisUI();
```
Comment (esterzhao) [inline leaf-ui/js/displayAnalysis.js:286]: much needed edit, thanks for this *in the distance, my branch cries in merge conflict*

File: leaf-ui/js/displayAnalysis.js [hunk 2]
```diff
@@ -297,7 +300,7 @@ function addNewAnalysisConfig(){
     // default Analysis Request needed for now for user assignments list
     // TODO: Look into perserving base UAL throughout analysisRequests
```
Comment (esterzhao) [inline leaf-ui/js/displayAnalysis.js:301]: looks like you fulfilled this TODO if I'm following correctly, in which case remove this comment

File: leaf-ui/js/objects.js [hunk 3]
```diff
@@ -239,13 +239,14 @@ class AnalysisConfiguration {
 	 * Updates Config Values from AnalysisRequest
 	 */
 	updateAnalysis(analysisRequest){
+		this.analysisRequest = analysisRequest;
```
Comment (esterzhao) [inline leaf-ui/js/objects.js:242]: git is such a spaz 🙄

File: leaf-ui/rappid-extensions/AnalysisInspector.js [hunk 4]
```diff
@@ -430,18 +430,23 @@ var AnalysisInspector = Backbone.View.extend({
...
-					selectElement.val(intEval.evaluationValue);
-				}
 				selectElement.append(options);
 				selectTd.append(selectElement);
 				row.append(selectTd);
-
 				}
 			$('#interm-list').append(row);
 		}
+		// Needed after looping through and appending rows to avoid voiding the select
```
Comment (esterzhao) [inline leaf-ui/rappid-extensions/AnalysisInspector.js:439]: can't say I understand this... def due to my unfamiliarity with this code but can you clarify (also in comments)

**Review by esterzhao — APPROVED** [review]
Looks good to me, no new unknown bugs found while testing. Would like to have @ibrodycalixto or someone else take a look before merging to confirm the IVT/Assignments modal works as intended since I am pretty shaky on those

File: leaf-ui/rappid-extensions/AnalysisInspector.js (hunk 4 above)
Comment (leshghi) [reply leaf-ui/rappid-extensions/AnalysisInspector.js:439]: Yes! Its just a weird way javascript working with html is set up which basically means even though before it was correctly setting the selectElement.val, that value wasn't propagating when adding selectElement into selectTd and then into the actual row. Instead we needed to build the row, and Then set the selected value inside that row using "this" to access the corresponding select menu.

File: leaf-ui/js/objects.js (hunk 3 above)
Comment (leshghi) [reply leaf-ui/js/objects.js:242]: trulyyyy

**Review by krhablutzel — COMMENTED** [review]
This looks great to me as well! I appreciate the function name change with `refreshAnalysisUI()`. Doesn't appear to add any new bugs

**Pull request merged**